internal/cmd/api-key/command_delete.go

```diff
@@ -28,17 +25,44 @@ func (c *command) newDeleteCommand() *cobra.Command {
 
 func (c *command) delete(cmd *cobra.Command, args []string) error {
 	c.setKeyStoreIfNil()
-	apiKey := args[0]
 
-	promptMsg := fmt.Sprintf(errors.DeleteResourceConfirmYesNoMsg, resource.ApiKey, apiKey)
-	if ok, err := form.ConfirmDeletion(cmd, promptMsg, ""); err != nil || !ok {
+	if err := c.confirmDeletion(cmd, args); err != nil {
 		return err
 	}
-	httpResp, err := c.V2Client.DeleteApiKey(apiKey)
+
+	deleteFunc := func(id string) error {
+		if r, err := c.V2Client.DeleteApiKey(id); err != nil {
+			return errors.CatchApiKeyForbiddenAccessError(err, deleteOperation, r)
+		}
+		return nil
+	}
+
+	deleted, err := resource.Delete(args, deleteFunc, c.postProcess)
+	resource.PrintDeleteSuccessMsg(deleted, resource.ApiKey)
+
 	if err != nil {
-		return errors.CatchApiKeyForbiddenAccessError(err, deleteOperation, httpResp)
+		return errors.NewErrorWithSuggestions(err.Error(), errors.APIKeyNotFoundSuggestions)
+	}
+	return nil
+}
+
+func (c *command) confirmDeletion(cmd *cobra.Command, args []string) error {
+	describeFunc := func(id string) error {
+		_, _, err := c.V2Client.GetApiKey(id)
+		return err
 	}
 
-	output.Printf(errors.DeletedResourceMsg, resource.ApiKey, apiKey)
-	return c.keystore.DeleteAPIKey(apiKey)
+	if err := resource.ValidateArgs(pcmd.FullParentName(cmd), args, resource.ApiKey, describeFunc); err != nil {
+		return err
+	}
+
+	if ok, err := form.ConfirmDeletionYesNo(cmd, form.DefaultYesNoPromptString(resource.ApiKey, args)); err != nil || !ok {
+		return err
+	}
```

Comment:
So if the user types "y", `ok` should be `true`, and we return `nil`, and if the user types "n", `ok` should be `false` and if `err == nil`, we should also return `nil`? I don't see what the difference is.

Reply:
This is an artifact leftover from a previous refactor. I'll change it.

Reply:
It was also a bug that deleted if users said `n`, so I'm going to add more integration tests to prevent it in the next commit.